families/seth/rpc/src/requests.rs

```diff
@@ -98,4 +108,55 @@ impl<S: MessageSender> ValidatorClient<S> {
 
         Ok(response)
     }
+
+    pub fn get_receipts(&mut self, block: &Block) -> Result<HashMap<String, SethTransactionReceipt>, String> {
+        let batches = &block.batches;
+        let mut transactions = Vec::new();
+        for batch in batches.iter() {
+            for txn in batch.transactions.iter() {
+                let header: TransactionHeader = match protobuf::parse_from_bytes(&txn.header) {
+                    Ok(h) => h,
+                    Err(_) => {
+                        continue;
+                    },
+                };
+                if header.family_name == "seth" {
+                    transactions.push(String::from(txn.header_signature.clone()));
+                }
+            }
+        }
+
+        let mut request = ClientReceiptGetRequest::new();
+        request.set_transaction_ids(protobuf::RepeatedField::from_vec(transactions));
+        let response: ClientReceiptGetResponse =
+            self.request(Message_MessageType::CLIENT_RECEIPT_GET_REQUEST, request)?;
+
+        let receipts = match response.status {
+            ClientReceiptGetResponse_Status::OK => response.receipts,
+            ClientReceiptGetResponse_Status::INTERNAL_ERROR => {
+                return Err(String::from("Received internal error from validator"));
+            },
+            ClientReceiptGetResponse_Status::NO_RESOURCE => {
+                return Err(String::from("Missing receipt"));
+            }
+        };
+        let mut seth_receipts: HashMap<String, SethTransactionReceipt> = HashMap::new();
+        for rcpt in receipts.iter() {
```

Comment:
You could consider using the Iterator function here, like `flat_map`, `filter`, `map` and `fold`